Mathlib/RingTheory/MvPolynomial/Symmetric.lean

```diff
@@ -281,12 +285,48 @@ theorem degrees_esymm [Nontrivial R] (n : ℕ) (hpos : 0 < n) (hn : n ≤ Fintyp
 
 end ElementarySymmetric
 
+section CompleteHomogeneousSymmetric
+
+open Classical Set
+
+/-- The `n`th complete homogeneous symmetric `MvPolynomial σ R`. -/
+def hsymm (n : ℕ) : MvPolynomial σ R := ∑ s : Sym σ n, (s.1.map X).prod
+
+lemma hsum_def (n : ℕ) : hsymm σ R n = ∑ s : Sym σ n, (s.1.map X).prod := rfl
+
+@[simp]
+theorem hsymm_zero : hsymm σ R 0 = 1 := by
+  simp only [hsymm, Finset.univ_unique, Sym.eq_nil_of_card_zero, Sym.val_eq_coe, Sym.coe_nil,
+    Multiset.map_zero, Multiset.prod_zero, Finset.sum_const, Finset.card_singleton, one_smul]
+
+@[simp]
+theorem hsymm_one : hsymm σ R 1 = ∑ i, X i := by
+  simp only [hsymm, Finset.univ_unique]
+  symm
+  apply Fintype.sum_equiv (Equiv.ofBijective _ Sym.one_bijective)
+  intro i
+  have : (Sym.one i).toMultiset = [i] := by rfl
```

Comment:
The idea is the following: if you inline a lemma twice or more, then split it off.

Reply:
Noted!